tests/IntegrationTests/WebAppUiTests/UiTestHelpers.cs

```diff
@@ -1,70 +1,275 @@
-        {
-            await entryBox.ClickAsync();
-            await entryBox.FillAsync(entryText);
-            await entryBox.PressAsync("Enter");
-        }
-
-        /// <summary>
-        /// This starts the recording of playwright trace files. The corresponsing EndAndWritePlaywrightTrace method will also need to be used.
-        /// This is not used anywhere by default and will need to be added to the code if desired
-        /// </summary>
-        /// <param name="page">The page object whose context the trace will record</param>
-        /// <returns></returns>
-        public static async Task StartPlaywrightTrace(IPage page) 
-        {
-            await page.Context.Tracing.StartAsync(new()
-            {
-                Screenshots = true,
-                Snapshots = true,
-                Sources = true
-            });
-        }
-
-        /// <summary>
-        /// This file gets written to the test's bin/debug/[dotnet version] folder
-        /// Use the app at https://trace.playwright.dev/ to easily view the trace, all trace data is opened locally by the web app.
-        /// </summary>
-        /// <param name="page">The page object whose context is recording a trace</param>
-        /// <returns>Nothing just creates the file</returns>
-        public static async Task EndAndWritePlaywrightTrace(IPage page) 
-        {
-            await page.Context.Tracing.StopAsync(new()
-            {
-                Path = "PlaywrightTrace.zip"
-            });
-        }
-    }
-}
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Management;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using Microsoft.Identity.Web.Test.Common;
+using Microsoft.IdentityModel.Tokens;
+using Microsoft.Playwright;
+using Microsoft.VisualStudio.TestPlatform.Utilities;
+using Xunit.Abstractions;
+
+namespace WebAppUiTests
+{
+    public static class UiTestHelpers
+    {
+        public static async Task FirstLogin_MicrosoftIdentityFlow_ValidEmailPassword(IPage page, string email, string password, ITestOutputHelper? output=null ,bool staySignedIn=false)
+        {
+            string staySignedInText = staySignedIn ? "Yes" : "No";
+
+            WriteLine(output, $"Logging in ... Entering and submitting user name: {email}");
+            ILocator emailInputLocator = page.GetByPlaceholder(TestConstants.EmailText);
+            await FillEntryBox(emailInputLocator, email);
+            await EnterPassword_MicrosoftIdentityFlow_ValidPassword(page, password, staySignedInText);
+        }
+
+        public static async Task SuccessiveLogin_MicrosoftIdentityFlow_ValidEmailPassword(IPage page, string email, string password, ITestOutputHelper? output = null, bool staySignedIn = false)

```

Comment:
>SuccessiveLogin_MicrosoftIdentityFlow_ValidEmailPassword( [](http://example.com/codeflow?start=33&length=57)

this is kind of long....would `SuccessiveLogin_ValidEmailPassword` work? same w/the others...really not set on either. :)